x-pack/plugins/apm/public/components/routing/telemetry_wrapper.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { useTrackPageview } from '../../../../observability/public';
+import { APMRouteDefinition } from '../../application/routes';
+import { redirectTo } from './redirect_to';
+
+export function TelemetryWrapper({
+  route,
+  props,
+}: {
+  route: APMRouteDefinition;
+  props: RouteComponentProps;
+}) {
+  const { component, render, path } = route;
+  const pathAsString = path as string;
+
+  useTrackPageview({ app: 'apm', path: pathAsString });
+  useTrackPageview({ app: 'apm', path: pathAsString, delay: 15000 });
+
+  if (component) {
+    return React.createElement(component, props);
+  }
+  if (render) {
+    return <>{render(props)}</>;
+  }
+  return <>{redirectTo('/')}</>;
+}
```

Comment:
is the `render` prob or the final default redirection to `/` being used anywhere in APM? I think i only see the `component` case explicitly used here.

Reply:
Sorry @ogupte, I didn't get what you meant.

Reply:
Sorry i was confusing 2 different components. I was looking at: https://github.com/elastic/kibana/pull/105297/files#diff-4128b0173be9d5028ac351674c344af919464dec16ca67728eb97e071c808b24R72 and saw that the `render` prop will never be set, and the `component` prop is always set, so I was asking if we really need to handle those cases, but i was conflating `TelemetryWrapper` with `ApmRoute`